### PR TITLE
SWAP-2039-upgrade-typescript-and-other-dependencies

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description

- All the dependencies are now using latest versions and Node v16 is used to build and run the app
- `.npmrc` file added and engines set in the `package.json` to prevent using earlier versions of node than 16
- alpine version upgraded in the `Dockerfile` to 3.15 which uses node 16 and chromium v93.

## Motivation and Context

Many of the dependencies weren't updated for some time and were lacking some versions behind latest.

## How Has This Been Tested

- e2e tests
- manual tests

## Fixes

https://jira.esss.lu.se/browse/SWAP-2039

## Changes

Before:

Screenshot_7

After:

Screenshot_9

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: src/__tests__/api/xlsx/sep-xlsx.spec.ts
Comment (jekabs-karklins) [inline src/__tests__/api/xlsx/sep-xlsx.spec.ts:10]: Couldn't this be just like this?
`beforeAll(() => {}, 15000);`

Comment (jekabs-karklins) [reply src/__tests__/api/xlsx/sep-xlsx.spec.ts:10]: Otherwise in Node 16 there is now
```
import { setTimeout } from 'timers/promises';

await setTimeout(5000)
```

**Review by jekabs-karklins — APPROVED** [review]

File: src/__tests__/api/xlsx/sep-xlsx.spec.ts
Comment (martin-trajanovski) [reply src/__tests__/api/xlsx/sep-xlsx.spec.ts:10]: It might be possible I will try it. Thanks for pointing this out.

Comment (martin-trajanovski) [reply src/__tests__/api/xlsx/sep-xlsx.spec.ts]: @jekabs-karklins I updated this and now it is using 
`
    beforeAll(async () => {
       await setTimeout(5000);
     }, 10000);
`

**Comment by mr-field** [comment]
I've tested this locally and opened https://github.com/UserOfficeProject/user-office-factory/pull/76 to make some small STFC-specific tweaks. @martin-trajanovski, could you have a look?

**Review by mr-field — CHANGES_REQUESTED** [review]
Other than the STFC-specific problems fixed in https://github.com/UserOfficeProject/user-office-factory/pull/76, this looks good. I'll approve this once I merge in my PR to avoid breaking the STFC setup.

**Pull request merged**
